code/datums/traits/negative.dm

```diff
@@ -320,6 +320,6 @@
 /datum/quirk/blindness/on_spawn()
 	var/mob/living/carbon/human/H = quirk_holder
 	var/obj/item/clothing/glasses/sunglasses/blindfold/white/glasses = new(get_turf(H))
-	if(!H.equip_to_slot(glasses, SLOT_GLASSES)) //if you can't put it on the user's eyes, put it in their hands, otherwise put it on their eyes
-		H.put_in_hands(glasses)
+	H.put_in_hands(glasses) //try to put it in the user's hands, then on the user's eyes
+	H.equip_to_slot(glasses, SLOT_GLASSES)
```

Comment:
It was correct before, you are just using the wrong proc

https://github.com/tgstation/tgstation/blob/1f805017d5d3ba2082905d1096b5f4bd3192c75d/code/modules/mob/mob.dm#L224-L227

Reply:
didnt the procs have return vars